bootstrap/src/main/java/io/airlift/bootstrap/Bootstrap.java

```diff
@@ -291,4 +306,28 @@ private static ColumnPrinter makePrinterForConfiguration(ConfigurationFactory co
         }
         return columnPrinter;
     }
+
+    @VisibleForTesting
+    static Map<String, String> replaceEnvironmentVariables(
+            Map<String, String> properties,
+            Map<String, String> environment,
+            BiConsumer<String, String> onError)
+    {
+        Map<String, String> replaced = new HashMap<>();
+        properties.forEach((propertyKey, propertyValue) -> {
+            Matcher matcher = ENV_PATTERN.matcher(propertyValue);
+            if (!matcher.matches()) {
+                replaced.put(propertyKey, propertyValue);
+                return;
+            }
+            String envName = matcher.group(1);
+            String envValue = environment.get(envName);
```

Comment:
Do empty environment variables show up as empty string or nulls?

Reply:
It should be possible for them to be empty, which is legal as a property value. Do you see any problems with allow this? It seems fine to me, although probably not useful.

Reply:
No, no problem with that. Just wanted to be sure they wouldn't be presented as `null` by `environment.get()`, which would result in them being considered missing.